jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/AzureFetchOrchestrator.java

```diff
@@ -63,6 +62,10 @@ public Map<String, Object> orchestrate() {
         try {
             List<SubscriptionVH> subscriptions = fetchSubscriptions();
             if (subscriptions.isEmpty()) {
+                if (numberOfAccounts > 0) {
+                    log.error("Error occurred in job azure-data-collector-job. Not able to connect to any of azure accounts.");
+                    AzureDiscoveryJob.jobStatus = "failed";
+                }
                 ErrorManageUtil.uploadError("all", "all", "all", "Error fetching subscription Info ");
                 return ErrorManageUtil.formErrorCode();
```

Comment:
another approach to all that without using `public static` from another class would be to add corresponding fail status param for this method. If you know that status must be `failed` then you could pass it as a parameter. Otherwise it will be calculated as usual by count of errors.
And make it `Failed` to be consistent - other two `Success` and `Partial success` are capitalized.

Reply:
i am setting jobStatus to failed in multiple places where the return type is void. this is the right way to check for this approach.
changing it to 'Failed' from 'failed'.